plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala

```diff
@@ -317,68 +321,76 @@ object ScalafmtPlugin extends AutoPlugin {
+    formatSources(
+      streams.value.cacheStoreFactory,
+      (unmanagedSources in scalafmt).value,
+      scalaConfig.value,
+      streams.value.log,
+      outputStreamWriter(streams.value),
+      fullResolvers.value
+    )
+  } tag (ScalafmtTagPack: _*)
+
+  private def scalafmtSbtTask = Def.task {
+    formatSources(
+      sbtSources.value.toSet,
+      sbtConfig.value,
+      streams.value.log,
+      outputStreamWriter(streams.value),
+      fullResolvers.value
+    )
+    formatSources(
+      metabuildSources.value.toSet,
+      scalaConfig.value,
+      streams.value.log,
+      outputStreamWriter(streams.value),
+      fullResolvers.value
+    )
+  } tag (ScalafmtTagPack: _*)
+
+  private def scalafmtCheckTask = Def.task {
+    val analysis = checkSources(
+      (scalafmt / streams).value.cacheStoreFactory,
+      (unmanagedSources in scalafmt).value,
+      scalaConfig.value,
+      streams.value.log,
+      outputStreamWriter(streams.value),
+      fullResolvers.value
+    )
+    trueOrBoom(analysis)
+  } tag (ScalafmtTagPack: _*)
+
+  private def scalafmtSbtCheckTask = Def.task {
+    trueOrBoom(
+      checkSources(
+        sbtSources.value,
         sbtConfig.value,
         streams.value.log,
         outputStreamWriter(streams.value),
         fullResolvers.value
       )
-      formatSources(
-        metabuildSources.value.toSet,
-        scalaConfig.value,
-        streams.value.log,
-        outputStreamWriter(streams.value),
-        fullResolvers.value
-      )
-    },
-    scalafmtCheck := {
-      val analysis = checkSources(
-        (scalafmt / streams).value.cacheStoreFactory,
-        (unmanagedSources in scalafmt).value,
+    )
+    trueOrBoom(
+      checkSources(
+        metabuildSources.value,
         scalaConfig.value,
         streams.value.log,
         outputStreamWriter(streams.value),
         fullResolvers.value
       )
-      trueOrBoom(analysis)
-    },
-    scalafmtSbtCheck := {
-      trueOrBoom(
-        checkSources(
-          sbtSources.value,
-          sbtConfig.value,
-          streams.value.log,
-          outputStreamWriter(streams.value),
-          fullResolvers.value
-        )
-      )
-      trueOrBoom(
-        checkSources(
-          metabuildSources.value,
-          scalaConfig.value,
-          streams.value.log,
-          outputStreamWriter(streams.value),
-          fullResolvers.value
-        )
-      )
-    },
+    )
+  } tag (ScalafmtTagPack: _*)
+
+  lazy val scalafmtConfigSettings: Seq[Def.Setting[_]] = Seq(
+    scalafmt := scalafmtTask.value,
+    scalafmtIncremental := scalafmt.value,
+    scalafmtSbt := scalafmtSbtTask.value,
+    scalafmtCheck := scalafmtCheckTask.value,
+    scalafmtSbtCheck := scalafmtSbtCheckTask.value,
     scalafmtDoFormatOnCompile := Def.settingDyn {
       if (scalafmtOnCompile.value) {
-        scalafmt in resolvedScoped.value.scope
+        (scalafmt in resolvedScoped.value.scope) tag(ScalafmtTagPack:_*)
```

Comment:
Is it necessary? `scalafmt` task already tagged with `ScalafmtTagPack`

Reply:
You're correct. Will remove.